Week-3/Problem 1.js

```diff
@@ -0,0 +1,47 @@
+// Memoize Function for addition of two Numbers
+var value=[];
+function add(value=[]){
+    var sum=0;
+    for(i =0;i<value.length;i++){
+        sum+=value[i];
+    }
+    console.log(`Sum of both numbers is ` + sum);
+    return (sum);
+}
+function memoize(fn){
+    const cache = new Map(); // Since cache is in pair of key and value
+    return function (...args){
+        const key = args.toString();
+        if(cache.has(key)){
+            console.log("Value Enter Previously Also");
+            return cache.get(key);
+        }
+        else{
+            cache.set(key,fn(...args));
+            console.log("Value Entered First Time");
+            return cache.get(key);
+        }
```

Comment:
This can better simplified as 
`if(!cache.has(key)){
            cache.set(key,fn(...args));
            console.log("Value Entered First Time");
        }
       return cache.get(key);`